cmd/root/bundle.go

```diff
@@ -108,19 +124,27 @@ func TryConfigureBundle(cmd *cobra.Command, args []string) error {
 	return configureBundle(cmd, args, bundle.TryLoad)
 }
 
-// environmentCompletion executes to autocomplete the argument to the environment flag.
-func environmentCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+// targetCompletion executes to autocomplete the argument to the target flag.
+func targetCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
 	b, err := loadBundle(cmd, args, bundle.MustLoad)
 	if err != nil {
 		cobra.CompErrorln(err.Error())
 		return nil, cobra.ShellCompDirectiveError
 	}
 
-	return maps.Keys(b.Config.Environments), cobra.ShellCompDirectiveDefault
+	return maps.Keys(b.Config.Targets), cobra.ShellCompDirectiveDefault
+}
+
+func initTargetFlag(cmd *cobra.Command) {
+	// To operate in the context of a bundle, all commands must take an "target" parameter.
+	cmd.PersistentFlags().String("target", "t", "bundle target to use (if applicable)")
```

Comment:
I tried running this but it doesn't work:
```
      --target string            bundle target to use (if applicable) (default "t")
```

You need to use `StringP` to include a shorthand. Now it's the default value.

Can you also add a test for this? Can be as simple as doing a `cmd.New()`, adding a bogus bundle command, and checking that the right flags are set / have the right values.